server/verify.go

```diff
@@ -138,33 +136,40 @@ func pubKeysEqual(k1 crypto.PublicKey, k2 crypto.PublicKey) bool {
 }
 
 // Checks if the provided AK public key can be trusted
-func checkAkTrusted(ak crypto.PublicKey, opts VerifyOpts) error {
-	if len(opts.TrustedAKs) == 0 {
-		return fmt.Errorf("no mechanism for AK verification provided")
+func checkAKTrusted(ak crypto.PublicKey, akCertBytes []byte, opts VerifyOpts) error {
+	checkPub := len(opts.TrustedAKs) > 0
+	checkCert := len(opts.TrustedRootCerts.Subjects()) > 0
+	if !checkPub && !checkCert {
+		return fmt.Errorf("no trust mechanism provided, either use TrustedAKs or TrustedRootCerts")
 	}
 
 	// Check against known AKs
-	for _, trusted := range opts.TrustedAKs {
-		if pubKeysEqual(ak, trusted) {
-			return nil
+	if checkPub {
+		for _, trusted := range opts.TrustedAKs {
+			if pubKeysEqual(ak, trusted) {
+				return nil
```

Comment:
I'm not sure if we expect a caller to ever provide both `checkPub` and `checkCert` or what such a caller should expect (boolean union or intersection of those checks)? Should we prefer to indent error flow here? https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88

Reply:
I fixed the `checkAKTrusted` to require that exactly one of `TrustedAKs` or `TrustedRootCerts` is used.

Thanks for the advice.